### DISCUSSION TITLE
WSL Storage Issue

### DISCUSSION THREAD

**Discussion by 0xusmanf** [Cyfrin/foundry-full-course-cu#228] [Q&A] [answered]
Hello,

I am using WSL with Windows 10 and facing a very weird problem, every time I restart my laptop the user-created files on WSL get deleted.

Can anybody please explain why this is happening and how I should resolve this issue?

**Comment by Shiva-Sai-ssb** [comment]
Hello @usmanfarooq91 , I believe the problem may be caused by an issue that occurred during the installation of WSL. My recommendation would be to try reinstalling WSL.

**Comment by 0xusmanf** [comment, marked as answer]
Hello @Shiva-Sai-ssb! thank you for your suggestion, I think I have figured out the solution. What happens when I start WSL is its storage location points to a mounted location on Windows like this `usmanfarooq90@DESKTOP-8OOB3I3:/mnt/c/Windows/system32$`, so I was redirecting it to this location `usmanfarooq90@DESKTOP-8OOB3I3:/mnt/wsl$` which is again a mounting storage location but also temporary, which is why my projects were getting deleted after restarting the laptop.

The correct solution is if your WSL starts with a mounted location like above then you have to `cd` your way back until it looks like this `usmanfarooq90@DESKTOP-8OOB3I3:/$` and then `cd` again to `home/"username"`. Make sure you use the username you created at the time of WSL setup. And after this, it will look like this `usmanfarooq90@DESKTOP-8OOB3I3:~$`.
